### DISCUSSION TITLE
What import alias would you like confiugred?

### DISCUSSION THREAD

**Discussion by patilharsh03** [vercel/next.js#46575] [Help] [answered]
### Summary

When I am creating the next app these options are coming to me. I never see this in any tutorial or any person who uses Next.js. can someone explain to me what it is? Is this a new thing on Next.js 13?? or I am only seeing these ??

![image](https://user-images.githubusercontent.com/58598919/219853094-63e3c208-e25e-4f52-a668-e643fd56f11e.png)

**Comment by jdfuturistic** [comment]
Hi, the solution for this problem is, put in the line command the next
`@component/*`
This in the case that you want adding an alias called component, also you can name of other form.

**Reply by KintuMagnus256** [reply]
Thanks you have save me

**Reply by Ingeregere** [reply]
Merci ça marche très bien.

**Reply by baregawi** [reply]
This is not a clear solution. I got:
```
npx create-next-app@latest @component/*
zsh: no matches found: @component/*
```
I created the directory and got the same.

**Comment by LaffertyDev** [comment, marked as answer]
I had the same question. I found this article by Abdulrahman Afaraetu to be helpful: https://dev.to/rhammy/path-aliases-in-nextjs-2fnc

![image](https://user-images.githubusercontent.com/3661934/221916390-3a740a0f-d242-4c40-aad3-50913cae0d56.png)

**Comment by ivanvinicius** [comment]
How can I skip this step and use relative path instead import alias?

**Reply by kbooth1000** [reply]
I think you could still use relative paths in your code and just ignore the "@/" or whatever if you want.

**Reply by patilharsh03** [reply]
Yes it's our choice.

**Comment by rwb196884** [comment]
So does anybody know what it means?

**Reply by Deeds67** [reply]
Yes, [look at the other answer in this thread](https://github.com/vercel/next.js/discussions/46575#discussioncomment-5153598). You can avoid doing this:

`import Layout from '../../components/Layout'`

and instead use a **Path alias** to do this:

`import Layout from '@/components/Layout'`

So it basically makes imports more concise/readable by having an alias for them.

**Comment by CodeVibranium** [comment]
So,
`@` here means `../../`
What if i'm in a component which is nested one more time ex `../../../`, then how would the `@` import alias would work?

**Reply by Californian** [reply]
Actually, the `@` translates to "whatever the root directory (or `src/` if you use it) is". So if you're two levels deep, it translates to `../../`; if you're three levels deep, it translates to `../../../`. This is so you don't have to think about imports relative to the file where you're importing, but rather relative to the root/src.

**Comment by maxswjeon** [comment]
I really think there should a way to skip this stage. For example, entering "none" for not creating it.

I use `baseUrl: "./src"` and use `@/*` as root route, `~/` for public asset route. It's annonying for creating it.

**Reply by tbabyuk** [reply]
If it doesn't allow you to skip this step, then just go ahead and say yes and complete the setup. Then, you can change this from your Next.js app by modifying the "jsconfig.json" or "tsconfig.json" file. To use regular import syntax, you can just leave the "paths" object blank, like so:

```
{
  "compilerOptions": {
    "paths": {}
  }
}
```
Keep in mind you might have to re-import a couple of imports that are already there to the new syntax to avoid an error. Hope this helps!

**Reply by khalidhassan10** [reply]
this one helped the most! so im basically referancing new aliases by using the `tsconfig.json` file and updating it so my code doesnt break when im for example renaming the defualt Pages.tsx!

**Comment by tannukannaojiya** [comment]
> ### Summary
> When I am creating the next app these options are coming to me. I never see this in any tutorial or any person who uses Next.js. can someone explain to me what it is? Is this a new thing on Next.js 13?? or I am only seeing these ??
> 
> ![image](https://user-images.githubusercontent.com/58598919/219853094-63e3c208-e25e-4f52-a668-e643fd56f11e.png)
> 
> ### Additional information
> _No response_
> 
> ### Example
> _No response_

This question solution is only type @/*  solve the problem.

**Reply by divyansh343** [reply]
i am in exact situation. what to prefer
![Uploading Screenshot 2023-11-26 195541.png…]()
